### DISCUSSION TITLE
Multi language support for templates

### DISCUSSION THREAD

**Discussion by Anan7Codes** [supabase/supabase#6160] [Questions] [answered]
Hey guys,

Is it possible to send our confirmation email, reset password, etc in different languages from the supabase templates?

**Comment by burggraf** [comment, marked as answer]
Here's an older related discussion I opened a while back with some possible ways to do this:
https://github.com/supabase/supabase/discussions/953

Also see: https://github.com/supabase/gotrue/issues/304

**Reply by burggraf** [reply]
You could also implement your own custom email system using a library I wrote here:
https://github.com/burggraf/supabase-mailer

**Reply by Anan7Codes** [reply]
Thanks Mark! I'll take a look at this.

**Comment by robthepaper** [comment]
Hi,
was looking for same feature and found a simpler way here: [https://supabase.com/docs/guides/auth/auth-email-templates](url)

For example when signup , you can add metadata like this : 
```
supabase.auth.signUp({
			email: email,
			password: password,
			options: {
				data: {
					domain: 'https://www.earlyaccess.trial.com',
					language: 'en'
				}
			}
		})
```

Then in the supabase backoffice, you can go to Email template and customize like this to access metadata passed earlier : 
```
<h2>Confirm your signup</h2>

<p>Follow this link to confirm your user:</p>
<p><a href="{{ .ConfirmationURL }}">Confirm your mail</a></p>

{{ .Data.language }}

{{ if eq .Data.language "en" }}
English
{{ else }}
Français
{{ end }}
```

Then in this exmaple, the email will display
English

**Reply by Oputo50** [reply]
@robthepaper  This only works for the email content. I don't think there is any available solution for translating the email subject which is defined in config.toml
